mongo/errors.go

```diff
@@ -76,11 +82,34 @@ func replaceErrors(err error) error {
 
 		return ce
 	}
+	if re, ok := err.(driver.ResponseError); ok {
+		return CommandError{
+			Name:    re.Message,
+			Wrapped: w,
+		}
+	}
+	if wce, ok := err.(driver.WriteCommandError); ok {
+		return WriteException{
+			WriteConcernError: convertDriverWriteConcernError(wce.WriteConcernError),
+			WriteErrors:       writeErrorsFromDriverWriteErrors(wce.WriteErrors),
+			Labels:            wce.Labels,
+		}
+	}
+	if wce, ok := err.(driver.WriteConcernError); ok {
+		return convertDriverWriteConcernError(&wce)
+	}
+	if we, ok := err.(driver.WriteError); ok {
+		return WriteError{
+			Index:   int(we.Index),
+			Code:    int(we.Code),
+			Message: we.Message,
+		}
+	}
 	if me, ok := err.(mongocrypt.Error); ok {
 		return MongocryptError{Code: me.Code, Message: me.Message}
 	}
 
-	return err
+	return fmt.Errorf("%s: %w", err.Error(), w)
```

Comment:
I'm a little unclear on our current Golang version; can we use `%w` (a 1.13 feature)? And, is this the way to wrap an error?